spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/config/DefaultQueryMappingConfiguration.java

```diff
@@ -0,0 +1,53 @@
+package org.springframework.data.jdbc.repository.config;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.springframework.data.jdbc.repository.QueryMappingConfiguration;
+import org.springframework.jdbc.core.ResultSetExtractor;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.lang.Nullable;
+import org.springframework.util.Assert;
+
+/**
+ * A {@link QueryMappingConfiguration} that allows for registration of {@link RowMapper}s and
+ * {@link ResultSetExtractor}s via a fluent Api.
+ *
+ * @author Jens Schauder
+ * @author Evgeni Dimitrov
+ */
+public class DefaultQueryMappingConfiguration implements QueryMappingConfiguration {
+
+	private Map<Class<?>, RowMapper<?>> mappers = new LinkedHashMap<>();
+
+	@Nullable
+	public <T> RowMapper<? extends T> getRowMapper(Class<T> type) {
+
+		Assert.notNull(type, "Type must not be null");
+
+		RowMapper<?> candidate = mappers.get(type);
+
+		if (candidate == null) {
+
+			for (Map.Entry<Class<?>, RowMapper<?>> entry : mappers.entrySet()) {
+
+				if (type.isAssignableFrom(entry.getKey())) {
+					candidate = entry.getValue();
```

Comment:
Is there any specification about ordering? If now, why not directly `return entry.getValue()`?